.github/scripts/predict-artifacts/predict.py

```
@@ -0,0 +1,373 @@
+
+def main() -> int:
+    ap = argparse.ArgumentParser()
+    ap.add_argument("--facts", default="facts.json")
+    ap.add_argument("--changed", required=True, help="file with one changed path per line")
+    ap.add_argument("--pr-title", default="")
+    ap.add_argument("--out-plan", default="plan.json")
+    ap.add_argument("--out-comment", default="comment.md")
+    args = ap.parse_args()
+
+    facts = load(args.facts)
+    changed = [l.strip() for l in Path(args.changed).read_text().splitlines() if l.strip()]
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)